data/helpers/conversations/upsertConversations.ts

```diff
@@ -77,10 +81,20 @@ const setupAndSaveConversations = async (
   });
 
   const conversationsToUpsert: Conversation[] = [];
-  conversations.forEach((conversation) => {
+  conversations.forEach(async (conversation) => {
     const alreadyConversationInDbWithTopic =
       alreadyConversationsByTopic[conversation.topic];
 
+    // If spam score is not computed, compute it
+    if (
+      conversation.spamScore === undefined ||
+      conversation.spamScore === null
+    ) {
+      await computeConversationsSpamScores(account, [
+        conversation as XmtpConversationWithUpdate,
+      ]);
```

Comment:
`computeConversationsSpamScores` can trigger a query to our db (for sender spam score)
not sure we should `await` here because if the network call fails we will not save the conversation to db!